File: Check-If-It-Is-a-Straight-Line.cc
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        int n = coordinates.size();
        
        if (n == 2) return true;

	if (coordinates[1][0] == coordinates[0][0]) {
	    int x = coordinates[0][0];
	    
	    for (int i = 1; i < n; i ++) if (coordinates[i][0] != x) return false;
	    return true;
	}
        
        float slope = float(coordinates[1][1] - coordinates[0][1])/(coordinates[1][0] - coordinates[0][0]);
        float c = coordinates[0][1] - slope*coordinates[0][0];
        
        // cout << slope << " " << c << endl;
         
        for (int i = 2; i < n; i ++) {
            if (abs(coordinates[i][1] - slope*coordinates[i][0] - c) <= pow(10, -9)) continue;
            else {
                // cout << abs(coordinates[i][1] - slope*coordinates[i][0]) << endl;
                return false;
            }
        }
        
        return true;
    }
};

File: Check-If-It-Is-a-Straight-Line.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
          if(coordinates.length>2)
        {
            
            float diffx = coordinates[0][0]-coordinates[1][0];
            float diffy = coordinates[0][1]-coordinates[1][1];
            
            //slope is infinite i.e. line is parallel to y-axis
            if(diffx == 0) 
            {
                //for every point value of x coordinate should be same
                for(int idx = 2; idx<coordinates.length; idx++)
                {
                    if(coordinates[idx][0]!=coordinates[0][0]) return false;
                }
            }
            else
            {
                float slope = diffy/diffx;
                
                //for every two points slopes should be equal 
                for(int idx = 2; idx<coordinates.length; idx++)
                {
                    float dx = coordinates[idx][0]-coordinates[0][0];
                    float dy = coordinates[idx][1]-coordinates[0][1];
                             
                    if(slope != (dy/dx)) return false;
                }
            }
        }
        return true;
    }
}